### PR TITLE
8 kr cv checked date

### LINKED ISSUE #9: 8. Como usuario, quiero markar un item como 'comprado' en la lista de compras así la app puede aprender qué tan seguido compro diferentes items
AC:

* El usuario puede clickear un checkbox o un elemento similar para marcar un item de la lista como comprado
* El item debe ser mostrado como checkeado por 24 horas despueés de que la compra se realiza (para asumir que el usuario no va a necesitar el item por lo menos en 1 día después de comprarlo)

### PR DESCRIPTION
**Pull request by candrepa1**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

The component List was changed, an input type checkbox was added to each item. It has attributes checked and onChange. For the checked attribute, a function called isPassed24Hours is passed, this function is in charge of checking if it has been 24 hours since the user checked the box of an item, after the 24 hours pass, the checkbox will automatically uncheck. For the onChange attribute, a function called handleCheck is passed, this function is in charge of updating the lastDate attribute on firebase, only when checked is true.

## Related Issue

closes #9 

## Acceptance Criteria

[ x ] The user can click on a checkbox or a similar element from the list to mark them as purchased.
[ x ] The displayed item should be checked for 24 hours after the purchase is made (to assume the users are not gonna need the item for at least one day).

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### After

 
Captura de pantalla de 2021-04-28 16-07-11

## Testing Steps / QA Criteria

From your terminal, pull down the branch
Run npm install to install the dependencies
Run npm start to view the features in the browser
Input a shared code, when you are redirected to the list view, try and check one of the items.

### CODE REVIEW

File: src/components/List/List.jsx [hunk 1]
```diff
@@ -5,20 +5,45 @@ import { useFirebase } from '../../hooks/useFirebase';
...
 
-  const { getAll } = useFirebase();
+  const { getAll, update } = useFirebase();
 
   const firebasePath = getAll().doc(token).collection('items');
 
   const [value, loading, error] = useCollection(firebasePath);
 
+  const handleCheck = (id, lastDate) => {
+    if (isPassed24Hours(lastDate) === false) {
```
Comment (MikeBLambert) [inline src/components/List/List.jsx]: Nice! I love how you're not allowing the timer to be reset here. Very clever.

File: src/components/List/List.jsx [hunk 2]
```diff
@@ -5,20 +5,45 @@ import { useFirebase } from '../../hooks/useFirebase';
...
+      const date = new Date();
+      update(token, id, { lastDate: date });
+    }
+  };
+
+  const isPassed24Hours = (date) => {
+    if (date) {
+      const currentDate = new Date();
+      const itemDate = new Date(date.toDate());
+      itemDate.setDate(itemDate.getDate() + 1);
```
Comment (MikeBLambert) [inline src/components/List/List.jsx:25]: Well that was a lot simpler of a solution than I was expecting. Nice job!

File: src/components/List/List.jsx [hunk 3]
```diff
@@ -5,20 +5,45 @@ import { useFirebase } from '../../hooks/useFirebase';
...
   const [value, loading, error] = useCollection(firebasePath);
 
+  const handleCheck = (id, lastDate) => {
+    if (isPassed24Hours(lastDate) === false) {
+      const date = new Date();
+      update(token, id, { lastDate: date });
+    }
+  };
+
+  const isPassed24Hours = (date) => {
```
Comment (MikeBLambert) [inline src/components/List/List.jsx]: Small nit-picky grammar thing: `isPast24Hours` or `has24HoursPassed`

**Review by MikeBLambert — APPROVED** [review]
I'm super impressed by how simple, clean, and elegant your solution was here. Great work!

File: src/components/List/List.jsx [hunk 4]
```diff
@@ -5,20 +5,45 @@ import { useFirebase } from '../../hooks/useFirebase';
...
+    if (date) {
+      const currentDate = new Date();
+      const itemDate = new Date(date.toDate());
+      itemDate.setDate(itemDate.getDate() + 1);
+      // itemDate.setMinutes(itemDate.getMinutes() + 2); // In case you wanna test it
```
Comment (FauricioValencia) [inline src/components/List/List.jsx:26]: nice idea this comment

File: src/components/List/List.jsx [hunk 5]
```diff
@@ -5,20 +5,45 @@ import { useFirebase } from '../../hooks/useFirebase';
...
+
+      return !(currentDate >= itemDate);
+    }
+    return false;
+  };
```
Comment (FauricioValencia) [inline src/components/List/List.jsx:32]: can you add space here :D

**Review by FauricioValencia — APPROVED** [review]

File: src/components/List/List.jsx (hunk 2 above)
Comment (dupjpr) [reply src/components/List/List.jsx:25]: It took me a while to understand why I had not realized that they were using the methods of the date-fns library. Nice work, I think the timer is very nice.👍🏻

**Review by dupjpr — APPROVED** [review]

File: src/components/List/List.jsx (hunk 3 above)
Comment (candrepa1) [reply src/components/List/List.jsx]: thanks for the correction Mike, the change has been done 👍

File: src/components/List/List.jsx (hunk 5 above)
Comment (candrepa1) [reply src/components/List/List.jsx:32]: I added a space in line 29, not sure if that's what you meant 🤔

**Review by anamafla — APPROVED** [review]

**Pull request merged**